cognite_toolkit/_cdf_tk/client/api/extended_timeseries.py

```diff
@@ -0,0 +1,301 @@
+            return self._set_pending_ids(instance_id)  # type: ignore[arg-type]
+        else:
+            raise TypeError(
+                "instance_id must be a NodeId, a tuple of (str, str), or a sequence of PendingIdentifier objects."
+            )
+
+    def _set_pending_ids(self, identifiers: Sequence[PendingInstanceId]) -> ExtendedTimeSeriesList:
+        """Set multiple pending identifiers for time series.
+
+        Args:
+            identifiers: A sequence of PendingIdentifier objects containing the pending instance IDs and optional IDs or external IDs.
+
+        Returns:
+            ExtendedTimeSeriesList: A list of ExtendedTimeSeries objects with the updated pending identifiers.
+        """
+        tasks = [
+            {
+                "url_path": f"{self._RESOURCE_PATH}/set-pending-instance-ids",
+                "json": {
+                    "items": [identifier.dump(camel_case=True) for identifier in id_chunk],
+                },
+                "api_subversion": "alpha",
+            }
+            for id_chunk in split_into_chunks(list(identifiers), 1000)
+        ]
+        tasks_summary = execute_tasks(
+            self._post,
+            tasks,
+            max_workers=self._config.max_workers,
+            fail_fast=True,
+        )
+        tasks_summary.raise_compound_exception_if_failed_tasks(
+            task_unwrap_fn=unpack_items_in_payload,
+        )
+
+        retrieved_items = tasks_summary.joined_results(lambda res: res.json()["items"])
+
+        return ExtendedTimeSeriesList._load(retrieved_items, cognite_client=self._cognite_client)
+
+    def retrieve(
+        self, id: int | None = None, external_id: str | None = None, instance_id: NodeId | None = None
+    ) -> ExtendedTimeSeries | None:
+        """`Retrieve a single time series by id. <https://developer.cognite.com/api#tag/Time-series/operation/getTimeSeriesByIds>`_
+
+        Args:
+            id (int | None): ID
+            external_id (str | None): External ID
+            instance_id (NodeId | None): Instance ID
+
+        Returns:
+            TimeSeries | None: Requested time series or None if it does not exist.
+
+        Examples:
+
+            Get time series by id:
+
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.time_series.retrieve(id=1)
+
+            Get time series by external id:
+
+                >>> res = client.time_series.retrieve(external_id="1")
+        """
+        identifiers = IdentifierSequence.load(ids=id, external_ids=external_id, instance_ids=instance_id).as_singleton()
+        return self._retrieve_multiple(
+            list_cls=ExtendedTimeSeriesList,
+            resource_cls=ExtendedTimeSeries,
+            identifiers=identifiers,
+            api_subversion="alpha",
+        )
+
+    def retrieve_multiple(
```

Comment:
This is a simple overwriting of the `.retrieve_multiple` method in the TimeSeriesAPI to return `ExtendedTimeSeries` and use the `alpha` flag.